bee-message/src/output/mod.rs

```diff
@@ -201,6 +204,43 @@ impl Output {
             Self::Nft(output) => Some(output.immutable_feature_blocks()),
         }
     }
+
+    /// Checks if a sufficient storage deposit was made for the given [`Output`].
+    pub fn check_sufficient_storage_deposit(&self, config: &ByteCostConfig) -> Result<(), Error> {
+        let maybe_deposit_condition = match self {
+            #[cfg(feature = "cpt2")]
+            Output::SignatureLockedSingle(_) | Output::SignatureLockedDustAllowance(_) => return Ok(()),
+            Output::Treasury(_) => return Ok(()), // `TreasuryOutput`s don't have `UnlockConditions`.
+            Output::Basic(_) | Output::Alias(_) | Output::Foundry(_) | Output::Nft(_) => self
+                .unlock_conditions()
+                .and_then(|conds| conds.get(StorageDepositReturnUnlockCondition::KIND))
+        };
+
+        let required = minimum_storage_deposit(config, self);
+
+        let maybe_deposit =
+            if let Some(UnlockCondition::StorageDepositReturn(storage_deposit_condition)) = maybe_deposit_condition {
+                Some(storage_deposit_condition.amount())
+            } else {
+                None
+            };
+
+        if self.amount() >= required {
+            match maybe_deposit {
+                Some(deposit) if deposit >= required => Ok(()),
+                Some(deposit) => Err(Error::InsufficientStorageDepositReturnAmount { required, deposit }),
+                None => Ok(()),
+            }
+        } else {
```

Comment:
Sorry if I maybe explained it wrong, but the amount always needs to be >= required, independent of `StorageDepositReturn` UnlockCondition
So in this case we always need to return 
```Rust
Err(Error::InsufficientStorageDepositAmount {
           required,
           amount: self.amount(),
})
```
And then additionally we need to check if there is such an unlock condition
```Rust
if let Some(UnlockCondition::StorageDepositReturn(storage_deposit_condition)) = maybe_deposit_condition {
    if storage_deposit_condition.amount() < required {
        return Err(Error::InsufficientStorageDepositReturnAmount { required, storage_deposit_condition.amount() })
    }
}
```

Reply:
https://github.com/lzpap/tips/blob/master/tips/TIP-0018/tip-0018.md#additional-syntactic-transaction-validation-rule
This also needs to be checked if there is a StorageDepositReturn

Reply:
https://github.com/iotaledger/tips/pull/38#discussion_r818507694

Reply:
So you need to add the check where you compare the amount in the StorageDepositReturn with the calculated storage deposit for a basic output with the address type from the StorageDepositReturn